source/common/router/router.cc

```diff
@@ -250,9 +250,15 @@ void Filter::chargeUpstreamCode(uint64_t response_status_code,
                                 const Http::HeaderMap& response_headers,
                                 Upstream::HostDescriptionConstSharedPtr upstream_host,
                                 bool dropped) {
+  uint64_t status_code_from_response_headers = Http::Utility::getResponseStatus(response_headers);
+  absl::optional<Grpc::Status::GrpcStatus> grpc_status =
+      Grpc::Common::getGrpcStatus(response_headers);
+  if (grpc_status.has_value()) {
+    status_code_from_response_headers = Grpc::Utility::grpcToHttpStatus(grpc_status.value());
+  }
   // Passing the response_status_code explicitly is an optimization to avoid
   // multiple calls to slow Http::Utility::getResponseStatus.
-  ASSERT(response_status_code == Http::Utility::getResponseStatus(response_headers));
+  ASSERT(response_status_code == status_code_from_response_headers);
```

Comment:
Why do we even compute ^^ if this is true for gRPC as well?

Reply:
Done. Removed the change here.

The goal is to keep the consistency. Since we replace the HTTP status code with gRPC status code inside onUpstreamHeaders, it is consistent to uses one status code instead of using different 2 status code within this function. In addition, stats is ignoring the grpc-status as well.

 Agree we only do it for the outlier detector here.